apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Tests/Trace/CloudTraceExtensionTest.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Diagnostics.Common;
+using Google.Cloud.Diagnostics.Common.Tests;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Internal;
+using Moq;
+using System;
+using Xunit;
+
+using ProjectId = Google.Cloud.Diagnostics.AspNetCore.CloudTraceExtension.ProjectId;
+using TraceProto = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Tests.Trace
+{
+    public class CloudTraceExtensionTest
+    {
+        private const string projectId = "pid";
+        private const string traceId = "105445aa7843bc8bf206b12000100f00";
+        private const ulong spanId = 81237123;
+
+        /// <summary>
+        /// Creates an <see cref="IServiceProvider"/> with all the needed dependencies to
+        /// create an <see cref="IManagedTracer"/>.
+        /// </summary>
+        /// <param name="context">The trace header context.</param>
+        /// <param name="rateLimiterShouldTrace">True if the rate limiter should allow tracing.</param>
+        private Mock<IServiceProvider> CreateProviderForCreateManagedTracer(
+            TraceHeaderContext context, bool rateLimiterShouldTrace)
+        {
+            Mock<IConsumer<TraceProto>> mockConsumer = new Mock<IConsumer<TraceProto>>();
+            Mock<IServiceProvider> mockProvider = new Mock<IServiceProvider>();
```

Comment:
Do we actually need a mock for this? Can't we just use a normal implementation and set it up appropriately? (The consumer makes more sense.)